### PR TITLE
Fix a long standing bug preventing the use of transparent views

### LINKED ISSUE #1165: post-processing pass doesn't blend transparent Views
**Describe the bug**
My understanding of the behaviour for View::setClearTargets possibly combined with View::setRenderTarget is that it should be possible to call the Renderer::render more than once, with a different camera projection for example, without overriding the previous render. 
One use case might be  in AR to render a camera frame background using an orthographic projection, followed by one or more object renderings using perspective projection. The following code should, as I understand, allow such rendering:
~~~~
  backgroundView->setClearTargets(true, true, false);
   view->setClearTargets(false, true, false);
   view->setRenderTarget(filament::View::TargetBufferFlags::DEPTH_AND_STENCIL);
 if (renderer->beginFrame(swapChain))
   {
      renderer->render(backgroundView);
      renderer->render(view);
      renderer->endFrame();
      return true;
   }
~~~~
with backgroundView configured with a different camera and scene.

When doing this I get only the foreground view with the foreground view clearColor in the background.  Commenting out either the view or the background view render does produce the correct individual rendering. Setting View::setRenderTarget  to  DEPTH_AND_STENCIL or NONE does not seem to make a difference.

**To Reproduce**
Source code to reproduce is available at:

[https://github.com/donaldmunro/FilamentTmp](https://github.com/donaldmunro/FilamentTmp)

**Expected behavior**
The background view should not be cleared during the second call to Renderer::render. 

**Screenshots**
See attached

**Desktop (please complete the following information):**
 - OS: Linux
 - GPU:  NVidia RTX 2070 and NVidia GeForce GTX 1060 Mobile (NVidia 418.56-10)
 - Backend: Both OpenGL and Vulkan.
background-only
combined
foreground-only

### PR DESCRIPTION
**Pull request by pixelflinger**
Fix #1165

In order to use transparent views, post-processing had to be disabled
because we were not able to blit post-processed buffers with blending.
This is now fixed by simply reverting to a quad.

A side effect of this, is improved performance when a scaling blit is
needed when MSAA is active.

This also introduced new "bugs", or at least weird behaviors: the
system needs to know when blending is needed, and this has to be
based on heuristics (unless we add a new api). Currently the heuristic
is that the COLOR buffer is not discarded and the view is cleared with 
an alpha value (or not cleared).

### CODE REVIEW

File: filament/src/Renderer.cpp [hunk 1]
```diff
@@ -259,7 +256,21 @@ void FRenderer::renderJob(ArenaScope& arena, FView& view) {
 
     FrameGraph fg(engine.getResourceAllocator());
 
-    const TextureFormat hdrFormat = getHdrFormat(view);
+    // Figure out if we need to blend this view into the framebuffer. Maybe this should be
+    // explicit, but since we don't have an API right now, we use heuristics:
+    // - we are keeping the color buffer before rendering, and
+    // - we are not clearing or clearing with alpha
+    // FIXME: make this an explicit API
+    const bool blending = !(view.getDiscardedTargetBuffers() & TargetBufferFlags::COLOR)
```
Comment (romainguy) [inline filament/src/Renderer.cpp]: We do have an explicit API for transparent rendering, it's a flag on the SwapChain. We should also check for clearColor.a < 1.0 even if the flag is set though.

Comment (pixelflinger) [reply filament/src/Renderer.cpp]: Actually the swapchain API only tells us that the buffer we're writing into is transparent, not that we should bend into it (which involves a full buffer blend) -- e.g. you might just want to clear the view with 0,0,0,0 and then draw opaque things in it.

Conversely, you could have an opaque swapchain and want to draw several views in it on top of each other, with blending.

So we need to know for sure that the user intended to blend the view as a whole.  Currently, I'm using the state of the clearcolor + discardflags to infer that.

The clearColor alpha check is done just one line below (line 265).

Comment (romainguy) [reply filament/src/Renderer.cpp]: Ah right I forgot about the use case of views blended on top of an opaque framebuffer. I saw the check for clearColor, that's why I mentioned it when talking about the flag.

File: filament/src/materials/blit.mat [hunk 2]
```diff
@@ -0,0 +1,30 @@
+material {
+    name : blit,
+    parameters : [
+        {
+            type : sampler2d,
+            name : color,
+            precision: high
```
Comment (romainguy) [inline filament/src/materials/blit.mat]: Why is the sampler precision set to high? We use at most RGBA16F which fits in `mediump`. UVs have to use `highp`, but not the samplers themselves.

File: filament/src/materials/blit.mat [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+vertex {
+    void postProcessVertex(inout PostProcessVertexInputs postProcess) {
+        postProcess.vertex.xy = postProcess.normalizedUV;
+    }
+}
+
+fragment {
+    void postProcess(inout PostProcessInputs postProcess) {
+        vec2 uv = variable_vertex.xy; // interpolated to pixel center
```
Comment (romainguy) [inline filament/src/materials/blit.mat]: It should be `HIGHP vec2 uv = …` since it's UVs.

File: filament/src/materials/blit.mat (hunk 2 above)
Comment (pixelflinger) [reply filament/src/materials/blit.mat]: good catch

File: filament/src/materials/blit.mat (hunk 3 above)
Comment (pixelflinger) [reply filament/src/materials/blit.mat]: yes!

File: filament/src/materials/blit.mat [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+vertex {
+    void postProcessVertex(inout PostProcessVertexInputs postProcess) {
+        postProcess.vertex.xy = postProcess.normalizedUV;
+    }
+}
+
+fragment {
+    void postProcess(inout PostProcessInputs postProcess) {
+        highp vec2 uv = variable_vertex.xy; // interpolated to pixel center
```
Comment (romainguy) [inline filament/src/materials/blit.mat]: We usually use our macro `HIGHP` instead of `highp`, but that works too.

**Comment by romainguy** [comment]
Can you update the release notes file too please?

**Comment by kpeeters** [comment]
Is this supposed to work on Linux? Rendering transparent views with `setPostProcessingEnabled(false)` on top of each other works fine, but as soon as there is a view which has PP turned on, it erases everything that has been rendered already.

**Pull request merged**